lib/shared-class.js

```diff
@@ -225,7 +276,11 @@ SharedClass.prototype.disableMethod = function(fn, isStatic) {
  */
 
 SharedClass.prototype.getKeyFromMethodNameAndTarget = function(name, isStatic) {
-  return (isStatic ? '' : 'prototype.') + name;
+  var addPrototypeString = !/^prototype\./.test(name);
```

Comment:
Interesting. So IIUC, this is adding support for `getKeyFromMethodNameAndTarget('prototype.name')`. I am wondering where is it used, why do we need to make this change?